cmd/snapshot/app/snapshot.go

```diff
@@ -0,0 +1,101 @@
+// Copyright © 2018 The Gardener Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/gardener/etcd-backup-restore/pkg/snapshot/snapshotter"
+	"github.com/gardener/etcd-backup-restore/pkg/snapstore"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	envStorageContainer = "STORAGE_CONTAINER"
+	defaultLocalStore   = "default.etcd.bkp"
+	backupFormatVersion = "v1"
+)
+
+var logger = logrus.New()
+
+// configuration is struct to hold configuration for utility
+type configuration struct {
+	schedule        string
+	etcdEndpoints   string
+	storageProvider string
+}
+
+// NewCommandStartSnapshotter create cobra command for snapshot
+func NewCommandStartSnapshotter(stopCh <-chan struct{}) *cobra.Command {
+	config := &configuration{}
+	var command = &cobra.Command{
```

Comment:
why dont you keep this as a variable in the package and initialize the flags in init() and in the function return the instance?

Reply:
First concern was to pass stopCh.
Regarding init flags, I wanted to read the parameters directly  in configuration object. I'm expecting configuration object will have more members later on, mostly specific to storage providers authentication info. In that case, we might wanna go with global config file option in future. We can surely keep config object also public. But I preferred this way, untill we think of config file, environment variables and may be something like viber to manage it in future.

Reply:
1. As with stopCh. We can create a channel in this command context as well instead of passing from main.go.
2. The configuration parameters are passed as flags to the command which makes it specific to this command ryt? So why do we need a global entitiy?